sensor/common/registry/registry_store.go

```diff
@@ -392,3 +392,221 @@ func (rs *Store) GetMatchingCentralRegistryIntegrations(imgName *storage.ImageNa
+	fromDefaultSA := serviceAcctName == defaultSA
+
+	for registryAddress, dce := range dockerConfig {
+		registryAddr := strings.TrimSpace(registryAddress)
+
+		if fromDefaultSA {
+			// We assume that registries found in the dockercfg secret managed by OCP for the default
+			// service account only references hostnames for the OCP internal registry.
+			rs.addClusterLocalRegistryHost(registryAddr)
+			if err := rs.upsertRegistry(namespace, registryAddr, dce); err != nil {
+				log.Errorf("Unable to upsert registry %q into store: %v", registryAddr, err)
+			}
+			continue
+		}
+
+		if env.DelegatedScanningDisabled.BooleanSetting() {
+			// If delegated scanning is disabled then we do not store additional secrets outside of those needed
+			// for scanning images from the OCP internal registry.
+			continue
+		}
+
+		if serviceAcctName != "" {
+			// Ignore secrets for service accounts other than default so that the
+			// default registry is not overwritten in the store.
+			continue
+		}
+
+		var err error
+		if isGlobalPullSecret {
+			err = rs.upsertGlobalRegistry(registryAddr, dce)
+		} else {
+			err = rs.upsertRegistry(namespace, registryAddr, dce)
+		}
+		if err != nil {
+			log.Errorf("unable to upsert registry %q into store: %v", registryAddr, err)
+		}
+	}
+}
+
+func (rs *Store) upsertSecretByName(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	isGlobalPullSecret := namespace == openshiftConfigNamespace && secretName == openshiftConfigPullSecret
+	hasBoundServiceAccount := serviceAcctName != ""
+
+	// To avoid partial upserts - hold the lock until the entire secret upserted.
+	rs.storeMutux.Lock()
+	defer rs.storeMutux.Unlock()
+
+	for registryAddress, dce := range dockerConfig {
+		registryAddr := strings.TrimSpace(registryAddress)
+
+		if hasBoundServiceAccount {
+			// We assume that registries found in dockercfg secrets managed by OCP only
+			// reference hostnames for the OCP internal registry.
+			rs.upsertPullSecretByNameNoLock(namespace, secretName, registryAddr, dce)
+			rs.addClusterLocalRegistryHost(registryAddr)
+			continue
+		}
+
+		if env.DelegatedScanningDisabled.BooleanSetting() {
+			// If delegated scanning is disabled then we do not store additional secrets outside of those needed
+			// for scanning images from the OCP internal registry.
+			continue
+		}
+
+		if isGlobalPullSecret {
+			if err := rs.upsertGlobalRegistry(registryAddr, dce); err != nil {
+				log.Errorf("Upserting global registry for pull secret %q, namespace %q, address %q: %v", secretName, namespace, registryAddr, err)
+			}
+		}
+
+		rs.upsertPullSecretByNameNoLock(namespace, secretName, registryAddr, dce)
+	}
+
+	log.Debugf("Upserted %d entries from secret %q in namespace %q", len(dockerConfig), secretName, namespace)
+}
+
+func (rs *Store) upsertPullSecretByNameNoLock(namespace, secretName, registryAddr string, dce config.DockerConfigEntry) {
+	registryAddr = urlfmt.TrimHTTPPrefixes(registryAddr)
```

Comment:
what happens in a weird situation where we get https://registry.com and http://registry.com? Seems like we overwrite the creds? Is that expected?

Reply:
We would overwrite the creds in that case yes. I would highly doubt within the same secret `https://<reg>`. `http://<reg>`, and/or `<reg>` would have different credentials intentionally.

We have ROX-25709 for taking these addresses to the 'next step' - which is to respect the path and glob patterns, thinking the 'scheme' trimming should be revisited as part of that.

I'm inclined to think since this wasn't 'technically' new logic with this PR - we defer the 'fix' when looking into the path/glob ticket. What do you think?